File: FangZhiHu/app/src/main/java/dagger/component/ActivityComponent.java
package dagger.component;

import com.tongqinyuan.fangzhihu.MainActivity;

import dagger.Component;
import dagger.module.ActivityModule;
import dagger.scope.ActivityScope;

/**
 * Created by tongqinyuan on 2018/3/8.
 */
@ActivityScope
@Component(dependencies = AppComponent.class)
public interface ActivityComponent {

    void inject(MainActivity activity);

}

File: FangZhiHu/KotlinOKhttp/src/main/java/com/kotlinokhttp/dagger/component/ActivityComponent.kt
package com.kotlinokhttp.dagger.component

import com.kotlinokhttp.MainActivity
import com.ytq.dagger.component.AppComponent
import com.ytq.dagger.scope.ActivityScope
import dagger.Component

/**
 * Created by tongqinyuan on 2018/3/9.
 */
@Component(dependencies = arrayOf(AppComponent::class))
@ActivityScope
interface ActivityComponent {

    fun inject(mainActivity: MainActivity)
}